src/state/apis/nft/utils.ts

```diff
@@ -57,57 +58,46 @@ export const openseaNetworkToChainId = (network: SupportedOpenseaNetwork): Chain
 export const chainIdToOpenseaNetwork = (chainId: ChainId): SupportedOpenseaNetwork | undefined =>
   CHAIN_ID_TO_OPENSEA_NETWORK_MAP[chainId]
 
-export const parseAlchemyNftContractToCollectionItem = (
-  contract: NftContract,
-  chainId: ChainId,
-): NftCollectionType => {
-  const { name, openSea } = contract
+export const updateNftItem = (originalItem: NftItem, currentItem: NftItem) => {
+  if (!originalItem.medias.length && currentItem.medias.length) {
+    originalItem.medias = currentItem.medias
+  }
+  if (originalItem.rarityRank === null && typeof currentItem.rarityRank === 'number') {
+    originalItem.rarityRank = currentItem.rarityRank
+  }
+  if (!originalItem.collection.floorPrice && currentItem.collection.floorPrice) {
+    originalItem.collection.floorPrice = currentItem.collection.floorPrice
+  }
 
-  const socialLinks = [
-    ...(openSea?.twitterUsername
-      ? [
-          {
-            name: 'Twitter',
-            label: 'Twitter',
-            url: `https://twitter.com/${openSea.twitterUsername}`,
-            logoUrl: '',
-          },
-        ]
-      : []),
-    ...(openSea?.discordUrl
-      ? [
-          {
-            name: 'Discord',
-            label: 'Discord',
-            url: openSea.discordUrl,
-            logoUrl: '',
-          },
-        ]
-      : []),
-    ...(openSea?.externalUrl
-      ? [
-          {
-            name: 'Website',
-            label: 'Website',
-            url: openSea.externalUrl,
-            logoUrl: '',
-          },
-        ]
-      : []),
-  ]
-  const id = toAssetId({
-    assetReference: contract.address,
-    assetNamespace: contract.tokenType.toLowerCase() as AssetNamespace,
-    chainId,
-  })
+  if (!originalItem.description && currentItem.description) {
+    originalItem.description = currentItem.description
+  }
 
-  return {
-    id,
-    chainId,
-    name: name || '',
-    floorPrice: openSea?.floorPrice ? bnOrZero(openSea.floorPrice).toString() : '',
-    openseaId: openSea?.collectionName || '',
-    description: openSea?.description || '',
-    socialLinks,
+  if (!originalItem.name && currentItem.name) {
+    originalItem.name = currentItem.name
   }
+
+  return originalItem
+}
+
+export const getAlchemyNftData = async (accountIds: AccountId[]): Promise<{ data: NftItem[] }> => {
```

Comment:
Can or should an sdk making requests be wrapped in the rtk query apis like the other ones?

Reply:
This was a conscious decision not to, most of the benefits we get from it is typings (caching is a non-problem because it's an RTK query consuming an RTK query), the wrapping `getNftUserTokens` will cache things for us. Felt like writing a wrapper just for the sake of writing a wrapper

Reply:
thanks for the breakdown, makes sense